api_docs/construct-narrow.md

```diff
@@ -51,7 +51,15 @@ important optimization when fetching messages in certain cases (e.g.
 when [adding the `read` flag to a user's personal
 messages](/api/update-message-flags-for-narrow)).
 
-**Changes**: In Zulip 9.0 (feature level 250), support was added for
+**Changes**: In Zulip 9.0 (feature level 263), narrows gained support
+for a new `with` operator, designed to be used in links that should
+follow a topic across being moved in a single API request. This operator
+takes in message ID as operand. If the message -- stream/private recipient
+type corresponding to the message ID exists and can be accessed by the
+user, then the narrow terms till the `with` operator is updated to point
+to the narrow containing the message.
```

Comment:
This detailed description of what the `with` operator means and what it's intended for should go in the main description of the operator, in the "Message IDs" section below. Then this "Changes" entry can just say something like:

> **Changes**: In Zulip 9.0 (feature level 263), the `with` operator was introduced.

In general, the principle with "Changes" entries is that if you just want to understand how the API works today, with the latest version of the server, you can ignore the "Changes" entries completely and just read the rest of the description. And then if you want to understand some previous version of the API, you start by reading up on the current version, then start reading the "Changes" entries to go back through time and get diffs that tell you about the previous versions.

So in particular when something didn't exist at a previous version, the "Changes" entry doesn't need to include any details about it — the only new information you need for understanding that previous version is the fact that such-and-such didn't exist yet.

(This is different from the central API changelog file. That one is meant to be read *forward*, not backward, for a reader who had some idea of a previous API to catch up on what's changed since then. So when a new feature is introduced, it tends to have a few more words about what it means or what it's for, although the details are still left for the feature's own documentation.)

Reply:
updated!